src/openms/include/OpenMS/FORMAT/PercolatorOutfile.h

```diff
@@ -0,0 +1,113 @@
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Hendrik Weisser $
+// $Authors: Hendrik Weisser $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_FORMAT_PERCOLATOROUTFILE_H
+#define OPENMS_FORMAT_PERCOLATOROUTFILE_H
+
+#include <OpenMS/KERNEL/MSExperiment.h>
+#include <OpenMS/METADATA/PeptideIdentification.h>
+#include <OpenMS/METADATA/ProteinIdentification.h>
+
+#include <boost/regex.hpp>
+
+#include <map>
+#include <vector>
+
+namespace OpenMS
+{
+  /**
+     @brief Class for reading Percolator tab-delimited output files.
+
+     For PSM-level output, the file extension should be ".psms".
+  */
+  class OPENMS_DLLAPI PercolatorOutfile
+  {
+
+  public:
+    
+    /// Types of Percolator scores
+    enum ScoreType { SCORE, QVALUE, POSTERRPROB };
+
+    /// Constructor
+    PercolatorOutfile();
+
+    /// Loads a Percolator output file
+    void load(const String& filename, ProteinIdentification& proteins, 
+              std::vector<PeptideIdentification>& peptides, 
+              enum ScoreType output_score = QVALUE, 
+              const String& psm_regex = "", bool count_from_zero = false,
+              const MSExperiment<>* experiment = 0);
+
+  private:
+
+    /// Information about a single fragment spectrum
+    struct ScanInfo
+    {
+      Int charge;
+      double rt, mz;
+    };
+
+    /// Mapping: spectrum index -> fragment spectrum details
+    typedef std::map<Size, struct ScanInfo> ScanInfoMap;
+
+    /// Description of how to extract information from PSM IDs
+    struct PSMInfoExtractor
+    {
+      boost::regex re; // regular expression for meta data extraction
```

Comment:
> Or am I overlooking something?

Not really. It also seemed more straight-forward that way.
How much of a problem is it to have the Boost dependency in the header?